core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.cpp

```diff
@@ -66,13 +67,39 @@ void OpenMPTargetInternal::fence(const std::string& name,
   }
 }
 int OpenMPTargetInternal::concurrency() const {
-  return 128000;  // FIXME_OPENMPTARGET
+  int max_threads = 2048 * 80;
+#if defined(KOKKOS_IMPL_ARCH_NVIDIA_GPU)
+  int max_threads_block = 2048;
+#if defined(KOKKOS_ARCH_AMPERE86)
+  max_threads = max_threads_block * 1536;
+#elif defined(KOKKOS_ARCH_AMPERE80)
+  max_threads = max_threads_block * 108;
+#elif defined(KOKKOS_ARCH_VOLTA72)
+  max_threads = max_threads_block * 84;
+#elif defined(KOKKOS_ARCH_VOLTA70)
+  max_threads = max_threads_block * 80;
+#elif defined(KOKKOS_ARCH_PASCAL60) || defined(KOKKOS_ARCH_PASCAL61)
+  max_threads = max_threads_block * 60;
+#endif
+#elif defined(KOKKOS_ARCH_INTEL_GPU)
+#pragma omp target map(max_threads)
+  { max_threads = omp_get_num_procs(); }
+
+  // Multiply the number of processors with the SIMD length.
+  max_threads *= 32;
+#endif
+
+  return max_threads;
 }
 const char* OpenMPTargetInternal::name() { return "OpenMPTarget"; }
 void OpenMPTargetInternal::print_configuration(std::ostream& os,
                                                bool /*verbose*/) const {
   // FIXME_OPENMPTARGET
   os << "Using OpenMPTarget\n";
+#if defined(KOKKOS_OPENEMPTARGET_HIERARCHICAL_INTEL_GPU)
```

Comment:
I guess you want
```suggestion
#if defined(KOKKOS_OPENMPTARGET_HIERARCHICAL_INTEL_GPU)
```
here and elsewhere.

Reply:
Make sure it is a KOKKOS_IMPL_OPENMPTARGET_ macro (implementation only)